### PR TITLE
more user input sanitation

### PR DESCRIPTION
**Pull request by matmair**
This PR:

    Changes all APIs that inherit from the generic classes to use built in shims as a base - this will enabe us to make changes to them easier
    Adds a new mixin on all api views that change data to clean the user input using mozillas bleach
    Updates all custom create() and update() functions in the API views to only use cleaned user data

<a href="https://gitpod.io/#https://github.com/inventree/InvenTree/pull/3204"></a>

### CODE REVIEW

File: InvenTree/InvenTree/mixins.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+        if getattr(instance, '_prefetched_objects_cache', None):
+            # If 'prefetch_related' has been applied to a queryset, we need to
+            # forcibly invalidate the prefetch cache on the instance.
+            instance._prefetched_objects_cache = {}
+
+        return Response(serializer.data)
+
+    def clean_data(self, data: dict) -> dict:
+        """Clean / snatize data.
```
Comment (SchrodingersGat) [inline InvenTree/InvenTree/mixins.py:38]: "sanitize"

File: InvenTree/order/api.py [hunk 2]
```diff
@@ -114,7 +116,7 @@ class PurchaseOrderList(APIDownloadMixin, generics.ListCreateAPIView):
 
     def create(self, request, *args, **kwargs):
         """Save user information on create."""
-        serializer = self.get_serializer(data=request.data)
+        serializer = self.get_serializer(data=self.clean_data(request.data))
```
Comment (SchrodingersGat) [inline InvenTree/order/api.py:119]: Nice work catching this, I was concerned that any custom calls to `create` may have been missed.

**Review by SchrodingersGat — APPROVED** [review]
Very nice work @matmair - a clean implementation that will serve us well.

**Comment by matmair** [comment]
@SchrodingersGat ready to merge now!

File: InvenTree/order/api.py (hunk 2 above)
Comment (matmair) [reply InvenTree/order/api.py:119]: @SchrodingersGat I went through every single method called create or update and each line in those

File: InvenTree/InvenTree/mixins.py (hunk 1 above)
Comment (matmair) [reply InvenTree/InvenTree/mixins.py:38]: @SchrodingersGat can you fix this in #3205 ?

Comment (SchrodingersGat) [reply InvenTree/InvenTree/mixins.py:38]: Done

**Pull request merged**